Show the enrolment status on large event card

nikomakela opened the pull request:
HH-283

<img width="1232" alt="image" src="https://user-images.githubusercontent.com/389204/214861338-f329f405-698e-4fd5-b8e6-246f2e762be9.png">

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr182.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr182.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr182.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr182.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr182.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr182.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr182.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr182.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr182.dev.hel.ninja :laughing::tada::tada::tada:

karisal-anders commented on packages/components/src/components/domain/event/eventEnrolmentStatus/useEventEnrolmentStatus.tsx:
```diff
@@ -0,0 +1,30 @@
+  } else if (enrolmentEndTime && new Date(enrolmentEndTime) < now) {
+    return EnrolmentStatusLabel.enrolmentEnded;
+  }
+  return EnrolmentStatusLabel.enrollable;
```
Isn't `EnrolmentStatusLabel.enrollable` a combination of `enrollable` and `not_enrollable_at_all`? As the same value is returned also when e.g.:
 - remainingAttendeeCapacity is null/undefined and today is between enrolmentStartTime and enrolmentEndTime
 - remainingAttendeeCapacity is a non-zero number and both enrolmentStartTime and enrolmentEndTime are null/undefined

Examples of possible ways to make this clearer IMHO:
 - Renaming the enum value to make it clearer what it means
 - Just returning an empty string here and putting a comment about not showing anything for the enrollable case and the not enrollable at all case
 - Making another enum value for the not enrollable at all case and handling it in the logic and translating it also as empty

nikomakela replied:
added an option

nikomakela committed:
feat: show the enrolment status in carousel cards

HH-283

karisal-anders commented on packages/components/src/components/domain/event/eventEnrolmentStatus/useEventEnrolmentStatus.tsx:
```diff
@@ -0,0 +1,32 @@
+    return EnrolmentStatusLabel.enrolmentNotStartedYet;
+  } else if (enrolmentEndTime && new Date(enrolmentEndTime) < now) {
+    return EnrolmentStatusLabel.enrolmentEnded;
+  } else if (enrolmentStartTime && enrolmentEndTime) {
```
The source ticket's description says remainingAttendeeCapacity needs to be not null and larger than zero for enrolment to be considered available. To take this into account how about e.g. changing this to:
```typescript
} else if (enrolmentStartTime && enrolmentEndTime && remainingAttendeeCapacity > 0) {
```
?

nikomakela replied:
well, I can add that without the lt-statement, because otherwise I need to check the nullinesh also.

karisal-anders replied:
I was thinking that as the type is `remainingAttendeeCapacity?: number | null` and as `(undefined > 0) === false` and `(null > 0) === false` and the only remaining possibilities through the type system are of type `number` using `remainingAttendeeCapacity > 0` here would handle also the undefined and null possibilities. But I think this is good enough already as it may be that the data in itself actually has an invariant of the value never being negative. So I leave it up to you whether you want to take that edge case into account or not.

nikomakela replied:
so, I did use `else if (enrolmentStartTime && enrolmentEndTime && remainingAttendeeCapacity)` , which checks the both, the nullness and zeroness.

karisal-anders reviewed (approved):
LGTM, a possibility of refactoring somehow to reduce code duplication looks like an avenue of possible improvement in the future, don't know how easy that would be but simply looking that almost identical changes have been done to three apps it begs the question whether there could be some way to put that stuff somewhere common and use it from there, but anyway, not necessary to do in this PR.

nikomakela commented:
It was decided in a refinement meeting that this could be merged if it would have a feature flag. So, lets add one.

- [ ] Add a feature flag witch can be used to switch on and off the new feature.

nikomakela committed:
refactor: add getLinkArrowLabel to the common package

HH-283

nikomakela committed:
refactor: add a reusable common RCHC-config

nikomakela committed:
chore: add some missing app configs and env variables

Add the variables to the example file that are missing from env.example,
but are listed in the AppConfig, and the opposite.

Start using the appConfigs for venue details page.

nikomakela committed:
feat: add feature flag for for enrolment status usage

HH-283

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**$(upper(replace(parameters.proxy,'-', ' '))) is deployed to platta: https://venue-graphql-proxy-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://liikunta-pr182.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
**$(upper(replace(parameters.proxy,'-', ' '))) is deployed to platta: https://events-graphql-proxy-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __failed__ for https://tapahtumat-pr182.dev.hel.ninja :crying_cat_face::anger::boom::boom:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr182.dev.hel.ninja :laughing::tada::tada::tada:

nikomakela committed:
feat: show the enrolment status on large event card

HH-283

terovirtanen commented:
**Hobbies-Helsinki branch is deployed to platta: https://harrastukset-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**$(upper(replace(parameters.proxy,'-', ' '))) is deployed to platta: https://events-graphql-proxy-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Events-Helsinki branch is deployed to platta: https://tapahtumat-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**Sports-Helsinki branch is deployed to platta: https://liikunta-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
**$(upper(replace(parameters.proxy,'-', ' '))) is deployed to platta: https://venue-graphql-proxy-pr182.dev.hel.ninja** :rocket::rocket::rocket:

terovirtanen commented:
## TestCafe result is __success__ for https://harrastukset-pr182.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://liikunta-pr182.dev.hel.ninja :laughing::tada::tada::tada:

terovirtanen commented:
## TestCafe result is __success__ for https://tapahtumat-pr182.dev.hel.ninja :laughing::tada::tada::tada:
